### ISSUE TITLE
Commonly faced issues and their solutions

### ISSUE THREAD

**Issue by TanmayDoesAI** [KillianLucas/open-interpreter#164]
This is a mega issue, created for commonly faced issues by the community with their solutions,

We have a new version out, kindly update it using `pip install -U open-interpreter`, that should fix a few commonly faced issues

Please make sure that you have a python version>=3.10 (**and for now also <3.12 as some dependencies are facing issues with it**) by running `python --version` or `python3 --version` in your command prompt

If you see the following error
```
OSError: [WinError 10106] The requested service provider could not be loaded or
initialized
Error during installation with OpenBLAS: Command '['C:\\Users\\Jane\\AppData\\
Local\\Microsoft\\WindowsApps\\python.exe', '-m', 'pip', 'install',
'llama-cpp-python']' returned non-zero exit status 1.
```
or 
```
× Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
│ exit code: 1
╰─> [116 lines of output]
```
this is a probably a problem with installing `llama-cpp-python`, and the fix we have figured for it is:
- seperately install `llama-cpp-python` using `pip install llama-cpp-python` in your terminal
- if you get an error it would probably give you the link to microsoft visual studio, so you would have to install Visual Studio 2019/2022 Community Edition, and then try installing again. Instructions can be found in [Code-Llama on Windows](https://github.com/KillianLucas/open-interpreter/blob/main/docs/WINDOWS.md)
- once you have `llama-cpp-python` installed try `pip install open-interpreter` to try to install it again, it should work, if it doesn't then kindly reach us out on [discord](https://discord.gg/QHhqDR9u) or open an issue for the same, providing adequate details for the same.

**Comment by TanmayDoesAI** [comment]
if you wish to utilize your GPU on windows, and are unable to do so even after `[?] Use GPU? (Large models might crash on GPU, but will run more quickly) (Y/n): Y` , then kindly go through (this markdown)[https://github.com/KillianLucas/open-interpreter/blob/main/GPU.md] , and if you still face any issues reach us out on our discord.

**Comment by TanmayDoesAI** [comment]
If you see an error similar to `interpreter command not found`
This is an issue with your Python/pip Scripts directory not being in your PATH environment variable.

To find out what that path is, run the following command.
`python -c 'import os,sysconfig;print(sysconfig.get_path("scripts"))'`

Now [add that directory](https://gist.github.com/nex3/c395b2f8fd4b02068be37c961301caa7#file-path-md) to your PATH environment variable and restart your terminal.

**Comment by jordanbtucker** [comment]
If you see an error similar to the following.

```
openai.error.InvalidRequestError: The model gpt-4 does not exist or you do not have access to it.
```

Please ensure that you have [prepaid billing](https://help.openai.com/en/articles/8264644-what-is-prepaid-billing) set up on your Open AI acount.

**Comment by TanmayDoesAI** [comment]
#627 acknowledges the current problem with python 3.12 and provides a temporary solution to it.

**Comment by ericrallen** [comment]
You may encounter a version error like:

> could not find a version that satisfies the requirment open-interpreter

This generally means you aren’t using a supported Python version.

Upgrade to 3.10 or 3.11 or follow the [instructions for a temporary fix for 3.12](https://github.com/KillianLucas/open-interpreter/issues/164#issuecomment-1759951240).

**Comment by ericrallen** [comment]
Huge thanks to @shawnmittal for this solution to the `sqlite3` version issue for Windows Subsystem for Linux (WSL) and Ubuntu.

When trying to run the interpreter command from the terminal, the following error message pops up:

> Your system has an unsupported version of sqlite3. Chroma requires sqlite3 >= 3.35.0.

Here's the fix for WSL/Ubuntu 20.04:

1. `wget https://sqlite.org/2023/sqlite-autoconf-3430200.tar.gz`
2. `tar -xvf sqlite-autoconf-3430200.tar.gz`
3. `cd sqlite-autoconf-3430200`
4. `sudo apt-get install libreadline-dev`
5. `./configure`
6. `make`
7. `sudo apt-get purge sqlite3`
8. `sudo make install`
9. Add the following to your `~/.bashrc`:
   ```shell
   export PATH="/usr/local/bin:$PATH"
   ```
10. `source ~/.bashrc`
11. Verify the `sqlite3` version:
   ```shell
   sqlite3 --version
   ```

**Comment by MikeBirdTech** [comment]
Closing this stale issue. Please create a new issue for any remaining problems. Thanks!
